### DISCUSSION TITLE
Copying plug-ins from host to AudioGridder

### DISCUSSION THREAD

**Discussion by MM502** [apohl79/audiogridder#1094] [Ideas]
It would be amazing if there was some sort of way to quickly move a plug-in from the host to AudioGridder. When a project is started it is often more convenient to work within the DAW without the need to outsource processing. 

As a project progresses there comes a point when it's time to move the processing over to AG.

Currently there is no way I am aware of to take CPU intensive plugins and move them to AG. Obviously the DAW sees AG as its own plugin but perhaps there is a way to at least copy and paste the presets/settings over from a VST running natively to one hosted by AudioGridder? 

I am using AG specifically to outsource Acustica plug-ins which do not have an internal preset system that could be used to achieve the above workflow.

**Comment by apohl79** [comment]
I don't think thats possible.

**Reply by jaacMusic** [reply]
I think what MM502 is saying is it would be nice for the AG plugin to host VST (like Waves StudioRack), and those same VSTs that are hosted/installed on the AG server side could be triggered with a button if one decides to offload CPU processing from the local DAW system to the remote AG server.
It is a great idea as it allows local processing of those plugins in case of AG server failure. I have found myself reconfiguring VST settings when I can't reload an AG plugin instance in a track.
Again, this could be more of a feature request!

**Reply by rncwalker** [reply]
AG does do that, think of AG as a wrapper for your Plugins you use, if you switch from your AG DAW install to the AG remote Server install, all your presets/settings will be the same.

**Reply by jaacMusic** [reply]
I don't think you understood my point. I saying to have the VSTs loaded in the AG plugin and let it process on the local DAW CPU as an option or/and have a button to load the same plugin on the AG server for processing. This will require the AG plugin to tell the AG server what to load when offloading is requested from the AG plugin. 
Currently, VSTs loaded in the AG plugin will only work with those VSTs that are already loaded and installed in the AG server. If the AG server were to be unavailable for any reason, the only option would be to bypass the VST in the AG plugin chain.

**Reply by rncwalker** [reply]
Install AG on the DAW and another PC and you can just switch between the two and your settings will follow the changes between both systems! the VST's need to be on both systems. I think you misunderstood MM502 question. He just does not want to loose the VST settings.

He already has the VST's on both systems, just not using the VST's through AG on the DAW. If he does that, it will solve his problem.

Robert

**Reply by MM502** [reply]
My question (with the aim of requesting it as a feature) related to being able to offload instances of plugins running natively within the DAW to AG at a later stage (via drag'n'drop of a plugin from the DAW mixer into the AG plugin window). As @apohl79 said this is not possible.

One thing that does work is to turn off both Reaper's and AG's plugin GUI and then manually transfer the settings . Whether that means it could be possible to program a custom action to move an FX chain from Reaper to AG is beyond my skill set.

@rncwalker suggested the workaround of using AG on the DAW which solves the core of the problem for me.  Working this way also prevents the problem @jaacMusic is describing (if I understand his post correctly).

This leaves the issue of DAW/VST presets:
Many plugins have their own preset management so any custom presets need to be copied from the DAW to the AG Server If, like me, you have saved your presets within Reaper you are out of luck (but that's the nature of the beast).

**Comment by rncwalker** [comment]
I use Reaper and have a few FX always open for in track control. (Gain, VU meter & EQ) then Audio gridder is used for all other FX.

AG Server is installed on the DAW as well as the FX PC Server that also has AS Server.

Doing this I can switch from the DAE AG to the FX Server and all of the FX will follow. the change

I only need to set my FX on one or the other Server.

I also have the added benefit that one of my FX that always crashed my system, now no longer crashes using it in AG, and works great.


Robert

**Reply by MM502** [reply]
That sounds like a really clever workaround - I'll try it out!

**Comment by rncwalker** [comment]
I currently have AG Server on my DAW and 3 AG servers, that do nothing else that be a server.
Remember that you need to have the same plugin's on all servers if you can. That only works if the plugin allows you to put it on more that one machine.

My DAW has all plugin's
AG server #1 has my MIDI Instruments
AG server #2 has my Audio FX's
AG server #3 not setup at this moment


Robert

**Comment by MM502** [comment]
Currently I use AG only for Acustica plugins and have one server connected to the DAW via dedicated 10Gbe. I also use Reaper and will start running AG on the DAW in order to be able to switch servers at a later stage - this is going to be neat.
Luckily, Acustica Audio allows up to five authorisations per license so no problems on that front. 
May I ask why exactly you chose to put MIDI instruments onto their own AG server? Is that so you can run a tighter block size? I thought if anything latency would be increased as you get the added network penalty.

Is it time for a dedicated thread where users can discuss their individual setups and exchange ideas?
